src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft.WinFx.targets

```diff
@@ -400,6 +404,57 @@
 
   </Target>
 
+  <!--
+
+                ==========================================
+                  CreateTemporaryTargetAssemblyProject
+                ==========================================
+
+                Name : CreateTemporaryTargetAssemblyProject
+
+  -->
+
+   <Target Name="CreateTemporaryTargetAssemblyProject"
+               Condition="'$(_RequireMCPass2ForMainAssembly)' == 'true' " >
+
+       <CreateTemporaryTargetAssemblyProject 
+                CurrentProject="$(MSBuildProjectFullPath)"
+                GeneratedCodeFiles="@(_GeneratedCodeFiles)"
+                CompileTypeName="Compile"
+                ReferencePath="@(ReferencePath)"
+                ReferencePathTypeName="ReferencePath" >
+           <Output TaskParameter="TemporaryTargetAssemblyProjectName" PropertyName="_TemporaryTargetAssemblyProjectName"/>
+       </CreateTemporaryTargetAssemblyProject>
+
+      <Message Text="GenerateTemporaryTargetAssembyProject project is $(_TemporaryTargetAssemblyProjectName)" />
+
+  </Target>
+
+  <!--
+
+                ==========================================
+                   RestoreTemporaryTargetAssemblyProject
+                ==========================================
+
+                Name : RestoreTemporaryTargetAssemblyProject
+
+  -->
+
+   <Target Name="RestoreTemporaryTargetAssemblyProject"
+               Condition="'$(_RequireMCPass2ForMainAssembly)' == 'true' " >
+
+      <Message Text="RestoreTemporaryTargetAssemblyProject project is $(_TemporaryTargetAssemblyProjectName)" Condition="'$(MSBuildTargetsVerbose)' == 'true'" />
+
+      <RunProjectBuildTarget 
+          ProjectName="$(_TemporaryTargetAssemblyProjectName)"
+          BuildTarget="Restore"
```

Comment:
Use a pattern similar to that in   `<Target Name="_CompileTemporaryAssembly"  DependsOnTargets="BuildOnlySettings;ResolveKeySource;CoreCompile" />` and `       <_CompileTargetNameForLocalType Condition="'$(_CompileTargetNameForLocalType)' == ''">_CompileTemporaryAssembly</_CompileTargetNameForLocalType>` used in `GenerateTemporaryTargetAssembly`